src/main/kotlin/lotto/domain/LottoDispenser.kt

```diff
@@ -1,30 +1,44 @@
 package lotto.domain
 
+import lotto.Tokenizer
 import lotto.domain.model.Lotto
 import lotto.domain.model.LottoNumber
 import lotto.domain.model.Rank
 import lotto.domain.model.WinningNumbers
 
-class LottoDispenser(val amount: Int) {
+class LottoDispenser(val amount: Int, manualLottoNumberTextList: List<String> = emptyList()) {
 
     lateinit var ranks: List<Rank>
-    lateinit var bonusNumber: LottoNumber
-    lateinit var winningNumbers: WinningNumbers
-    val lottoList: List<Lotto> = makeLottoList()
+
+    val autoLottoList: List<Lotto> = makeLottoList(manualLottoNumberTextList.size)
+    val manualLottoList: List<Lotto> = makeManualLottoList(manualLottoNumberTextList)
+    val lottoList: List<Lotto>
+        get() = manualLottoList + autoLottoList
 
     init {
         require(amount >= MINIMUM_PRICE) { "구입 금액은 ${MINIMUM_PRICE}원 이하가 될 수 없습니다" }
     }
 
-    fun checkWinningLottoList() {
+    fun match(winningNumbers: WinningNumbers) {
         ranks = lottoList.map { lotto ->
-            Rank.win(winningNumbers, lotto, bonusNumber)
+            Rank.win(winningNumbers, lotto)
         }
     }
 
-    private fun makeLottoList(): List<Lotto> {
-        return List(amount / MINIMUM_PRICE) {
-            Lotto()
+    private fun makeLottoList(manualLottoCount: Int): List<Lotto> {
+        val total = amount / MINIMUM_PRICE
+        val autoLottoCount = total - manualLottoCount
+        return List(autoLottoCount) { Lotto() }
+    }
+
+    private fun makeManualLottoList(manualLottoNumberText: List<String>): List<Lotto> {
+        val manualLottoCount = manualLottoNumberText.size
+        return List(manualLottoCount) { index ->
+            val lottoNumbers = Tokenizer.tokenize(manualLottoNumberText[index])
+                .map { numberText ->
+                    LottoNumber(numberText.toInt())
+                }
+            Lotto(lottoNumbers)
```

Comment:
```suggestion
            Tokenizer.tokenize(manualLottoNumberText[index])
                .map { numberText ->
                    LottoNumber(numberText.toInt())
                }.let(::Lotto)
```
let과 메서드 참조를 활용하면, 불필요한 변수 없이 활용이 가능해요

Reply:
🙆‍♂️